com.unity.perception/Runtime/Randomization/Randomizers/RandomizerExamples/Utilities/GameObjectOneWayCache.cs

```diff
@@ -75,6 +100,9 @@ public void ResetAllObjects()
                     {
                         // Position outside the frame
                         obj.transform.localPosition = new Vector3(10000, 0, 0);
+                        var tags = obj.GetComponents<RandomizerTag>();
+                        foreach (var tag in tags)
+                            tag.Unregister();
```

Comment:
Calling getcomponents for all objects in each iteration might become a major performance issue if there are many objects. I wonder if it would be worthwhile to keep a dictionary of objects to their tag components in the tag manager as a cache?

Reply:
This also applies to the Register() for loop a few lines above. If the objects were fetched from cache (not newly created) we could look at our object to tag cache to speed things up instead of using getcomponents.

Reply:
I'm not sure if this optimization is necessary yet. We'd need to do a performance test to be sure.

Reply:
You know what, I'ma do it anyways.